torch_geometric/nn/models/lightgcn.py

```diff
@@ -0,0 +1,311 @@
+        assert (num_nodes >= 1), 'Number of nodes must be larger than 0.'
+        assert (embedding_dim >= 1), 'Embedding dimension must be larger ' \
+            'than 0.'
+        assert (objective in LightGCN.objectives), 'Objective must be one ' \
+            f'of {LightGCN.objectives}'
+
+        if alpha is not None:
+            assert (alpha.size(0) == (num_layers + 1)), 'alpha must be of ' \
+                'size (num_layers + 1).'
+        else:
+            alpha = [1 / (num_layers + 1)] * (num_layers + 1)
+
+        self.num_layers = num_layers
+        self.num_nodes = num_nodes
+        self.embedding_dim = embedding_dim
+        self.embeddings = Embedding(num_nodes, embedding_dim)
+        self.objective = objective
+        self.lambda_reg = lambda_reg
+        self.alpha = alpha
+        self.convs = ModuleList()
+
+        for _ in range(num_layers):
+            self.convs.append(LGConv())
+
+        if objective == LightGCN.objectives[0]:
+            self.loss_fn = BPRLoss(self.lambda_reg)
+        else:
+            self.loss_fn = BCELoss()
+
+    def reset_parameters(self):
+        self.embeddings.reset_parameters()
+
+    def forward(self, edge_index: Adj, edge_label_index: Adj = None,
+                weights: Tensor = None, **kwargs) -> Tensor:
+        """Computes rankings or link probabilities for pairs of nodes.
+
+        Args:
+            edge_index (Adj): Edge tensor specifying the connectivity of the
+                graph.
+            edge_label_index (Adj, optional): Edge tensor specifying the start
+                and end nodes for which to compute rankings or proabilities.
+                If :obj:`edge_label_index` is set to :obj:`None`, all edges in
+                :obj:`edge_index` will be used (default: :obj:`None`).
+            weights (Tensor, optional): Pre-computed embeddings or node
+                features which should be used to initialize the embedding
+                layer (default: :obj:`None`).
+        """
+        if edge_label_index is None:
+            edge_label_index = edge_index
+
+        if weights is not None:
+            assert (weights.shape == self.embeddings.shape), \
+                'Pre-computed embeddings weights must match shape of \
+                embedding layer.'
+            self.embeddings.weight.data.copy_(weights)
+
+        out = self._propagate_embeddings(edge_index)
+
+        nodes_fst = torch.index_select(out, 0, edge_label_index[0, :].long())
+        nodes_sec = torch.index_select(out, 0, edge_label_index[1, :].long())
+        out = torch.sum(nodes_fst * nodes_sec, dim=-1)
+
+        if self.objective == LightGCN.objectives[1]:
+            out = torch.sigmoid(out)
+
+        return out
+
+    def predict(self, edge_index: Adj, edge_label_index: Adj = None,
+                prob: bool = False) -> Tensor:
+        """Predict links between nodes specified in :obj:`edge_label_index`.
```

Comment:
nit: Since this method works only for the link prediction objective, I think it's worth specifying so here, in addition to the assertion. Perhaps the method name can become `predict_link`.

Reply:
This is a good idea! Would you mind changing it with the documentation?

Reply:
Done :)